class.jetpack.php

```diff
@@ -436,56 +436,68 @@ class_exists( 'Jetpack_Sitemap_Manager' )
 		delete_transient( self::$plugin_upgrade_lock_key );
 	}
 
-	static function update_active_modules( $modules ) {
-		$current_modules = Jetpack_Options::get_option( 'active_modules', array() );
-
-		$success = Jetpack_Options::update_option( 'active_modules', array_unique( $modules ) );
-
-		if ( is_array( $modules ) && is_array( $current_modules ) ) {
-			$new_active_modules = array_diff( $modules, $current_modules );
-			foreach( $new_active_modules as $module ) {
-				/**
-				 * Fires when a specific module is activated.
-				 *
-				 * @since 1.9.0
-				 *
-				 * @param string $module Module slug.
-				 * @param boolean $success whether the module was activated. @since 4.2
-				 */
-				do_action( 'jetpack_activate_module', $module, $success );
-
-				/**
-				 * Fires when a module is activated.
-				 * The dynamic part of the filter, $module, is the module slug.
-				 *
-				 * @since 1.9.0
-				 *
-				 * @param string $module Module slug.
-				 */
-				do_action( "jetpack_activate_module_$module", $module );
-			}
+	static function isArray( $arr ) {
+		if ( ! is_array( $arr ) ) {
+			return array();
+		}
+		return $arr;
+	}
 
-			$new_deactive_modules = array_diff( $current_modules, $modules );
-			foreach( $new_deactive_modules as $module ) {
-				/**
-				 * Fired after a module has been deactivated.
-				 *
-				 * @since 4.2.0
-				 *
-				 * @param string $module Module slug.
-				 * @param boolean $success whether the module was deactivated.
-				 */
-				do_action( 'jetpack_deactivate_module', $module, $success );
-				/**
-				 * Fires when a module is deactivated.
-				 * The dynamic part of the filter, $module, is the module slug.
-				 *
-				 * @since 1.9.0
-				 *
-				 * @param string $module Module slug.
-				 */
-				do_action( "jetpack_deactivate_module_$module", $module );
-			}
+	static function update_active_modules( $modules ) {
+		$current_modules      = Jetpack_Options::get_option( 'active_modules', array() );
+		$active_modules       = Jetpack::get_active_modules();
+	
+		$new_active_modules   = array();
+		$new_deactive_modules = array();
+		$current_modules      = ( is_array( $current_modules ) ) ? $current_modules : array();
+		$modules              = ( is_array( $modules ) ) ? $modules : array();
```

Comment:
I've taken a quick look at the usages of `::update_active_modules` and didn't spot a potential place where `$modules` wouldn't be an array. Could you please elaborate on the need for this one?

Reply:
I just carried forward the `is_array` checks from master branch, since I wasn't sure if it was kept for any special reason or backward compatibility. It does look like there isn't a chance for a non-array value, unless the option is somehow given a `null` value. Nevertheless, I've fully removed the `is_array` checks for now in [aa74795](https://github.com/Automattic/jetpack/pull/12270/commits/aa74795aa318c709427fac7ed93d16c7dd550674).